jax/lax/lax_control_flow.py

```diff
@@ -977,4 +975,130 @@ def _root_batch(args, dims, **params):
 root_p.def_abstract_eval(_root_abstract_eval)
 ad.primitive_jvps[root_p] = _root_jvp
 xla.initial_style_translations[root_p] = xla.lower_fun(_root_impl, initial_style=True)
-batching.primitive_batchers[root_p] = _root_batch
+batching.deftraced(root_p)
+
+
+_Solves = collections.namedtuple('_Solves', 'forward tangent cotangent')
+
+def linear_solve(matvec, b, forward_solve, tangent_solve=None,
+                 cotangent_solve=None, symmetric=False):
+  """Differentiably solve the linear map matvec(x)=b for x.
+
+  Required invariant:
+      x = solve(matvec, b)
```

Comment:
`solve` should be `forward_solve`. Perhaps also describe what `(co)tangent_solve` is supposed to be and what invariants it observes?

Reply:
Done, I wrote a full docstring.